### ISSUE TITLE
Can you please let me know when will be this update reflect in the puppet forge module?

### ISSUE THREAD

**Issue by moitrasayan007** [voxpupuli/puppet-splunk#230]
Can you please let me know when will be this update reflect in the puppet forge module?

**Comment by alexjfisher** [comment]
@moitrasayan007 Have you been able to test the current git version?  It was such a major refactor, any feedback would be welcome.  Thanks.

**Comment by moitrasayan007** [comment]
@alexjfisher , yeah I tried testing it but I am facing some issue, for example, 

Issue 1

when I am trying to include only ::splunk::enterprise::service::nix, it is saying that the service_name is undef for service.pp file.

Issue 2

I also tried installing the package and everything from 

class { '::splunk::enterprise':
      version                => '7.2.5.1',
      package_name           => 'splunk',
      enterprise_package_src => 'S3 yum repo',
      package_source         => 'S3 yum repo',
      package_provider       => 'yum',
      package_ensure         => 'installed',
      service_name           => 'splunk',
      service_file           => '/etc/init.d/splunk',
    }

it is taking by default taking the version as 7.2.4.2 as mentioned in params.pp.

Issue 3

I want to take the files like web.conf, serverclass.conf and other files by including ::splunk::params as type as I don't want all the files to be populated in all splunk servers but it looks like it is not allowing me to separately declare the files from ::splunk::params as the files are also created in `/enterprise/config.pp` and it places the default files in all servers which I want to restrict.

Please let me know if there are any solutions for these.

**Comment by alexjfisher** [comment]
Issue 1:
I think `splunk::enterprise::service` and `splunk::enterprise::service::nix` should be considered private (and documented as such).

Issue 2:
Maybe a bug or maybe just an oddity with this module. @nick-markowski - is this something you might be able to fix?  It's an odd pattern to expect users to declare params.pp

**Comment by nick-markowski** [comment]
`splunk::enterprise::service` and `nix` should be made private.

I'll try to replicate the instantiation issues and see if we can clean it up

**Comment by zhasaan** [comment]
Hi Nick,
I am also seeing the systemd service issue with "splunk" vs "SplunkForwarder" as service name issue after installing splunkforwarder-7.2.2 RPM. I am anxiously waiting on new release which seem to ahve fix of the issue#215 in order for me to move forward. 
===========================================================
Debug: Exec[license_splunkforwarder](provider=posix): Executing check '/usr/bin/test -f /opt/splunkforwarder/ftr'
  Debug: Executing with uid=splunk: '/usr/bin/test -f /opt/splunkforwarder/ftr'
  Debug: Exec[license_splunkforwarder](provider=posix): Executing 'splunk ftr --accept-license --answer-yes --no-prompt'
  Debug: Executing with uid=splunk: 'splunk ftr --accept-license --answer-yes --no-prompt'
  Notice: /Stage[main]/Splunk::Platform::Posix/Exec[license_splunkforwarder]/returns: executed successfully
  Info: /Stage[main]/Splunk::Platform::Posix/Exec[license_splunkforwarder]: Scheduling refresh of Service[splunk]
  Debug: /Stage[main]/Splunk::Platform::Posix/Exec[license_splunkforwarder]: The container Class[Splunk::Platform::Posix] will propagate my refresh event
  Debug: Exec[enable_splunkforwarder](provider=posix): Executing '/opt/splunkforwarder/bin/splunk enable boot-start -user splunk'
  Debug: Executing: '/opt/splunkforwarder/bin/splunk enable boot-start -user splunk'
  Notice: /Stage[main]/Splunk::Platform::Posix/Exec[enable_splunkforwarder]/returns: executed successfully
  Info: /Stage[main]/Splunk::Platform::Posix/Exec[enable_splunkforwarder]: Scheduling refresh of Service[splunk]
  Debug: /Stage[main]/Splunk::Platform::Posix/Exec[enable_splunkforwarder]: The container Class[Splunk::Platform::Posix] will propagate my refresh event
  Debug: Class[Splunk::Platform::Posix]: The container Stage[main] will propagate my refresh event
  Debug: Executing: '/usr/bin/systemctl is-active splunk'
  Debug: Executing: '/usr/bin/systemctl is-enabled splunk'
  Debug: Executing: '/usr/bin/systemctl unmask splunk'
  Debug: Executing: '/usr/bin/systemctl start splunk'
  Debug: Runing journalctl command to get logs for systemd start failure: journalctl -n 50 --since '5 minutes ago' -u splunk --no-pager
  Debug: Executing: 'journalctl -n 50 --since '5 minutes ago' -u splunk --no-pager'
  Debug: Executing: '/usr/bin/systemctl is-active splunk'
  Error: Systemd start for splunk failed!
  journalctl log for splunk:
  -- No entries --
  
  Error: /Stage[main]/Splunk::Virtual/Service[splunk]/ensure: change from stopped to running failed: Systemd start for splunk failed!
  journalctl log for splunk:
  -- No entries --
  Debug: /Stage[main]/Splunk::Virtual/Service[splunk]: Skipping restart; service is not running
==============================================================

Your help is greatly appreciated please.

**Comment by nick-markowski** [comment]
I have been away for the past month, I'll take a look at these issues as soon as I can
